### DISCUSSION TITLE
convert yaml array to string representation to export as environment variable

### DISCUSSION THREAD

**Discussion by ev0net** [mikefarah/yq#1899] [Q&A]
I use yq to add environment vars from yaml files.

example.yaml:

    var0: string0
    var1: string1
    ary0: ['aryval0', 'aryval1', 'aryval2']

code:

    while read -r key val; do
    
    	eval "export $key=${val}"
    	
    done < <(yq '.[] | key + " " + .' example.yaml)

error:

    !!seq (ary0) cannot be added to a !!str ()

desired outcome:

The same yaml file gets read into bash, environment variables, ansible and php at different times for different purposes.

I need to catch array entries when reading the yaml file and convert those array entries into a string representation to export as an environment variable, but I don't know the syntax or methods to do this in yq.

array declaration and values; I don't know how to get the environment variable example to work: 

    ansible/yaml: ary0: ['aryval0', 'aryval1', 'aryval2'] 
    bash: ary0=('aryval0', 'aryval1', 'aryval2') 
    php: $ary0 = array('aryval0', 'aryval1', 'aryval2'); 
    environment variable: ary0="aryval0:aryval1:aryval2"

Thanks!

**Comment by mikefarah** [comment]
You can  do it in a `yq` script file like this:

bash.yq
```
.[] |(
    ( select(kind == "scalar") | key + "='" + . + "'"),
    ( select(kind == "seq") | key + "=(" + (map("'" + . + "'") | join(",")) + ")")
)
```

Then run it with yq like:

```bash
yq --from-file=bash.yq example.yaml
```

To get:
```bash
var0='string0'
var1='string1'
ary0=('aryval0','aryval1','aryval2')
```

Explanation:
- `.[]` matches all top level elements
- Then we're going to have a string expression for each of the different types that will product the bash syntax, and use the union operator `,` to join them together
- Scalars, we just need the key and quoted value: `( select(kind == "scalar") | key + "='" + . + "'")`
- Sequences (or arrays) are trickier, we need to quote each value and `join` them with `,`: `map("'" + . + "'") | join(",")`

**Reply by ev0net** [reply]
Thank you so much for your reply @mikefarah !

I've tried different things with your above examples and some other examples from the yq docs like

`yq '(.. | select(kind == "scalar")) line_comment = "this is a scalar"' sample.yml`

but I'm getting the following types of errors regardless.

```
Error: 1:14: invalid input text "kind == \"scalar\"..."
Error: 2:14: invalid input text "kind == \"seq\") |..."
```

Any thoughts?

version: yq (https://github.com/mikefarah/yq/) version v4.35.2
